File: 223__Rectangle_Area.py
class Solution(object):
    def computeArea(self, A, B, C, D, E, F, G, H):
        """
        :type A: int
        :type B: int
        :type C: int
        :type D: int
        :type E: int
        :type F: int
        :type G: int
        :type H: int
        :rtype: int
        """
        total = (C - A) * (D - B) + (G - E) * (H - F)
        if A >= G or C <= E or B >= H or D <= F:
            return total
        dis_x = min(C, G) - max(A, E)
        dis_y = min(D, H) - max(B, F)
        return total - dis_x * dis_y

File: Java/223__Rectangle_Area.java
class Solution {
    public int computeArea(int A, int B, int C, int D, int E, int F, int G, int H) {
        long overlapX = Math.max(0, (long) Math.min(C, G) - (long) Math.max(A, E));
        long overlapY = Math.max(0, (long) Math.min(D, H) - (long) Math.max(B, F));
        long overlap = overlapX * overlapY;
        return (C - A) * (D - B) + (G - E) * (H - F) - (int) overlap;
    }
}
